fabric-tool-attribute-api-v1/src/main/java/net/fabricmc/fabric/impl/tool/attribute/VanillaToolsSupport.java

```diff
@@ -0,0 +1,61 @@
+package net.fabricmc.fabric.impl.tool.attribute;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.Blocks;
+import net.minecraft.block.Material;
+import net.minecraft.item.Item;
+import net.minecraft.tag.Tag;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+
+import net.fabricmc.api.ModInitializer;
+import net.fabricmc.fabric.api.tool.attribute.v1.FabricToolTags;
+
+public class VanillaToolsSupport implements ModInitializer {
+	private static final Logger LOGGER = LogManager.getLogger("fabric-tool-attribute-api-v1");
+
+	@Override
+	public void onInitialize() {
+		for (Block block : Registry.BLOCK) {
+			registerBlockSupport(Registry.BLOCK.getId(block), block);
+		}
+
+		registerSupport(FabricToolTags.PICKAXES, 3, Blocks.OBSIDIAN);
```

Comment:
this seems un-maintainable tbh

Reply:
The list looks rather arbitrary to me, why not use net.minecraft.item.AxeItem.EFFECTIVE_BLOCKS etc=

Reply:
Actually true, we still need to hardcode the blocks with special mining levels though

Reply:
@modmuss50 that's the thing - Mojang's system of hardcoding block mining levels and effectivenesses is fundamentally unmaintainable. We're doing what we can to work around it but if Mojang hardcodes it we have to too.

Reply:
We now work around this by using tool handlers, check the latest code and you will see that this is basically completely different